File: app/src/main/java/mo/dyna/sifomobileanalyticssdkforandroid/CategoryActivity.java
package mo.dyna.sifomobileanalyticssdkforandroid;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.NavUtils;
import android.support.v7.app.ActionBar;
import android.support.v7.app.ActionBarActivity;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.Toast;

import se.sifo.analytics.mobileapptagging.android.MobileTaggingFramework;

/**
 * Created by Peter on 2015-04-17.
 */
public class CategoryActivity extends ActionBarActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_category);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

        ListView listView = (ListView) findViewById(android.R.id.list);
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                sendTag((String) parent.getItemAtPosition(position));
            }
        });
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                NavUtils.navigateUpFromSameTask(this);
                overridePendingTransition(R.anim.slide_in_right, R.anim.slide_out_right);
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        overridePendingTransition(R.anim.slide_in_right, R.anim.slide_out_right);
    }

    private void sendTag(String category) {
        if (MobileTaggingFramework.getInstance() != null) {
            ApplicationImpl.tagInfo().setCategory(1, category);
            MobileTaggingFramework.getInstance().sendTag(ApplicationImpl.tagInfo().getCategories(), "", "");
            Toast.makeText(this, getString(R.string.toast_sent_tag, category), Toast.LENGTH_SHORT).show();

            startActivity(new Intent(this, ContentActivity.class));
            overridePendingTransition(R.anim.slide_in_left, R.anim.slide_out_left);
        }
    }


}

File: app/src/main/java/se/kantarsifo/mobileanalytics/sampleapp/native_view/CategoryActivity.kt
package se.kantarsifo.mobileanalytics.sampleapp.native_view

import android.content.Context
import android.content.Intent
import android.os.Bundle
import se.kantarsifo.mobileanalytics.framework.TSMobileAnalytics
import se.kantarsifo.mobileanalytics.sampleapp.base.BaseActivity
import se.kantarsifo.mobileanalytics.sampleapp.databinding.ActivityCategoryBinding
import se.kantarsifo.mobileanalytics.sampleapp.util.TagInfo

class CategoryActivity : BaseActivity() {

    private lateinit var binding: ActivityCategoryBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityCategoryBinding.inflate(layoutInflater)
        setContentView(binding.root)
        init()
    }

    override fun onBackPressed() {
        tryToFinishAffinity()
    }

    // Private methods

    private fun init() {
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        binding.list.setOnItemClickListener { parent, _, position, _ ->
            sendTag(parent.getItemAtPosition(position) as String)
        }
    }

    private fun sendTag(category: String) {
        if (TSMobileAnalytics.instance != null) {
            TagInfo.setCategory(1, category)
            TSMobileAnalytics.instance?.sendTag(
                    categories =  TagInfo.getCategories(),
                    contentID = ""
            )
            ContentActivity.start(this)
        }
    }

    companion object {
        fun start(context: Context) {
            val intent = Intent(context, CategoryActivity::class.java)
            context.startActivity(intent)
        }
    }

}
